### ISSUE TITLE
[Bug]: Muxing broken w/o using H.265 on macOS (after addition of H.265 enc. support in ffmpeg)

### ISSUE THREAD

**Issue by azec-pdx** [Puyodead1/udemy-downloader#160]
### What happened?

I pulled from `master` and was running the same workflow as before, with download command being:
```
# With/o H.265
python main.py \
  -c https://<REDACTED>.udemy.com/course/database-engines-crash-course \
  --download-assets --download-captions --disable-ipv6 \
  --lang en --log-level DEBUG
```

This led to an error being thrown as following:
![mux_process_broken_after_H 265_enc_support](https://github.com/Puyodead1/udemy-downloader/assets/1272830/baaa7c5a-e9c9-4c31-a870-733b6665eacd)

The problematic part appears to be use of the `-vtag hvc1` flag when `use_h265=False` (default value) and user still wants to encode with H.264. From the documentation on [libx265](https://trac.ffmpeg.org/wiki/Encode/H.265#FinalCutandApplestuffcompatibility) they say:
> To make your file compatible with Apple "industry standard" H.265 you have to add the following argument `-tag:v hvc1`.

This looks like parameters for H.265 were mixed in the command that copies video codec from the input and that leads to mismatching parameter set passed to `ffmpeg`. 

When I test with H.265 , everything is fine on the same platform, the speed of muxing just depends on parameters passed:
```
# With H.265
python main.py \
  -c https://<REDACTED>.udemy.com/course/database-engines-crash-course \
  --download-assets --download-captions --disable-ipv6 \
  --use-h265 --h265-crf 22 --h265-preset fast \
  --lang en --log-level DEBUG
```

So I believe there is regression introduced after the latest changes to support H.265 when not using H.265.

### Expected Result

The `ffmpeg` command should not have failed and command for `ffmpeg` should be further parametrized/improved when not using H.265 on macOS (and possibly other paltforms).

### Branch

master/main

### Relevant log output

```shell
Provided in the screenshot above
```

### Other information

Platform:

**macOS Version:**
```(Monterey, Version 12.6.5)```

**ffmpeg Version:**
```
ffmpeg version 6.0 Copyright (c) 2000-2023 the FFmpeg developers
  built with Apple clang version 14.0.0 (clang-1400.0.29.202)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/6.0 --enable-shared --enable-pthreads --enable-version3 --cc=clang --host-cflags= --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libaribb24 --enable-libbluray --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-librav1e --enable-librist --enable-librubberband --enable-libsnappy --enable-libsrt --enable-libsvtav1 --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libsoxr --enable-libzmq --enable-libzimg --disable-libjack --disable-indev=jack --enable-videotoolbox
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
Hyper fast Audio and Video encoder
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}...
```

**Comment by azec-pdx** [comment]
One thing I tried was reverting command that was used before libx265 support being added:

```
def mux_process(video_title, video_filepath, audio_filepath, output_path):
    """
    @author Jayapraveen
    """
    ############################### Worked with ffmpeg before H.265 codec support ###########################
    # if os.name == "nt":
    #     command = 'ffmpeg -nostdin -loglevel error -y -i "{}" -i "{}" -acodec copy -vcodec copy -fflags +bitexact -map_metadata -1 -metadata title="{}" "{}"'.format(
    #         video_filepath, audio_filepath, video_title, output_path)
    # else:
    #     command = 'nice -n 7 ffmpeg -nostdin -loglevel error -y -i "{}" -i "{}" -acodec copy -vcodec copy -fflags +bitexact -map_metadata -1 -metadata title="{}" "{}"'.format(
    #         video_filepath, audio_filepath, video_title, output_path)
    ##########################################################################################################

    codec = "hevc_nvenc" if use_nvenc else "libx265"
    transcode = "-hwaccel cuda -hwaccel_output_format cuda" if use_nvenc else []
    if os.name == "nt":
        if use_h265:
            command = 'ffmpeg {} -y -i "{}" -i "{}" -c:v {} -crf {} -preset {} -c:a copy -fflags +bitexact -map_metadata -1 -metadata title="{}" "{}"'.format(
                transcode, video_filepath, audio_filepath, codec, h265_crf,
                h265_preset, video_title, output_path)
        else:
            # command = 'ffmpeg -y -i "{}" -i "{}" -c:v copy -vtag hvc1 -c:a copy -fflags +bitexact -map_metadata -1 -metadata title="{}" "{}"'.format(
            #     video_filepath, audio_filepath, video_title, output_path)
            command = 'ffmpeg -nostdin -loglevel error -y -i "{}" -i "{}" -acodec copy -vcodec copy -fflags +bitexact -map_metadata -1 -metadata title="{}" "{}"'.format(
                video_filepath, audio_filepath, video_title, output_path)
    else:
        if use_h265:
            command = 'nice -n 7 ffmpeg {} -y -i "{}" -i "{}" -c:v libx265 -crf {} -preset {} -c:a copy -fflags +bitexact -map_metadata -1 -metadata title="{}" "{}"'.format(
                transcode, video_filepath, audio_filepath, codec, h265_crf,
                h265_preset, video_title, output_path)
        else:
            # command = 'nice -n 7 ffmpeg -y -i "{}" -i "{}" -c:v copy -vtag hvc1 -c:a copy -fflags +bitexact -map_metadata -1 -metadata title="{}" "{}"'.format(
            #     video_filepath, audio_filepath, video_title, output_path)
            # command = 'nice -n 7 ffmpeg -y -i "{}" -i "{}" -c:v copy -c:a copy -fflags +bitexact -map_metadata -1 -metadata title="{}" "{}"'.format(
            #     video_filepath, audio_filepath, video_title, output_path)
            command = 'nice -n 7 ffmpeg -nostdin -loglevel error -y -i "{}" -i "{}" -acodec copy -vcodec copy -fflags +bitexact -map_metadata -1 -metadata title="{}" "{}"'.format(
                video_filepath, audio_filepath, video_title, output_path)

    process = subprocess.Popen(command, shell=True)
    log_subprocess_output("FFMPEG-STDOUT", process.stdout)
    log_subprocess_output("FFMPEG-STDERR", process.stderr)
    ret_code = process.wait()
    if ret_code != 0:
        raise Exception("Muxing returned a non-zero exit code")

    return ret_code
```
But that leads to videos not playable at all (neither in VLC nor QuickTime).

**Comment by Puyodead1** [comment]
huh, not sure how that happened but that explains some other conversations I've seen related to hvc1. the command should be fixed in 1ee5d7966406091e45f72e5745840d30fdfe4b21
